developer/src/kmldmlc/src/kmldmlc.ts

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env node
+/**
+ * kmldmlc - Keyman LDML Keyboard Compiler
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as program from 'commander';
+
+import Compiler from './keyman/ldmlkeyboard/compiler';
+
+let inputFilename: string;
+
+const KEYMAN_VERSION = require("@keymanapp/keyman-version").KEYMAN_VERSION;
+
+/* Arguments */
+program
+  .description('Compiles Keyman LDML keyboards')
+  .version(KEYMAN_VERSION.VERSION_WITH_TAG)
+  .arguments('<infile>')
+  .action(infile => inputFilename = infile)
+  .option('-o, --outFile <filename>', 'where to save the resulting .kmx file');
+
+program.parse(process.argv);
+
+// Deal with input arguments:
+if (!inputFilename) {
+  exitDueToUsageError('Must provide a LDML keyboard .xml source file.');
```

Comment:
Will there be any validation of the inputFilename (like in keyboard ID)?

Reply:
I think filenames are fairly arbitrary from the POV of the compiler at this point. But linting is a big discussion that we will need have once we get some basic compilation going!